sechub-doc/src/docs/asciidoc/documents/shared/concepts/concept_templates_and_assets.adoc

```diff
@@ -0,0 +1,308 @@
+  `DELETE /api/administration/asset/$assetId`
+
+The REST API will always create an audit log entry 
+
+[IMPORTANT]
+====
+This works only when {sechub} storage is reused by {pds}.
+If the {pds} uses its own storage (which should NOT be done for production, but only
+for PDS solution development) the assets must be uploaded directly to the PDS storage 
+location! 
+====
+
+===== SecHub template and asset handling
+====== Validation
+When a {sechub} job starts and there is a template data definition inside the configuration, 
+{sechub} will validate if the project has a template assigned for the location inside
+the configuration. e.g. templates with type `webscan-login` may only be defined inside 
+web scan login configuration).
+
+If this validation fails, the complete {sechub} job will fail and stop processing.
+
+[[sechub-concept-asset-upload-lazy]]
+====== Upload in storage as cache lazily
+When validation did not fail, {sechub} will check if the asset is already uploaded to
+storage (S3/NFS) already.
+
+When not uploaded to storage, the file will be uploaded before the job is further processed.
+
+====== Template PDS parameter calculation
+If validation and lazy uploaded did not fail, {sechub} will calculate a
+{pds} parameter : `pds.template.metadata` with JSON (see parameter syntax for details).
+
+.PDS parameter syntax for template meta data 
+[source,json]
+----
+{
+    "template-metadata" : [{//<1>
+        {
+            "template" : "single-singon", //<2>
+            "type: "webscan-login",//<3>
+            "assets" : [ //<4>
+                {
+                    "id" : "asset-id-1", //<5>
+                    "sha256" : "434c6c6ec1b0ed9844149069d7d45ac18e72505b" //<6> 
+                }
+            ]
+        }]
+    }     
+}
+
+----
+<1> Meta data array
+<2> Template identifier - just as an information for logging etc.
+<3> Template type
+<4> Asset information array
+<5> ID of the asset, necessary for storage download
+<6> checksum of the asset ZIP file, necessary to check after download
+
+===== PDS asset handling
+{sechub} calls {pds}, with {pds} parameter `pds.template.metadata` (syntax is described above).
+
+The {pds} intance will fetch all wanted asset ZIP file for the current product 
+from storage (S3 or NFS) and extract it to `$workspaceFolder/assets/`. 
+
+Before extraction is done a checksum for the downloaded ZIP file is created and compared
+with the checksum from template meta data. If it is different the {pds} job will fail
+with a dedicated error message.
+
+If checksum wer evalid, the assets ZIP file will be unzipped below a subfolder with the 
+template type:
+ 
+[source,text]
+----
+`$workspaceFolder/
+     assets/
+        $template-type/
+                    ...
+----
+
+An example:
+
+The `MY_PRODUCT.zip` file contains
+
+[source,text]
+----
+/script.js
+/development/
+             debug-settings.json
+----
+
+and the template meta data looks like this:
+
+[source,text]
+----
+{
```

Comment:
This JSON differs from the example in line 194ff. (e.g. template-metadata is a list there)
I suggest to omit the JSON code here and refer to 194ff.

Reply:
I will apply the changes later while implementing the parameter model data.